src/promptflow-evals/promptflow/evals/evaluate/_code_client/code_client.py

```diff
@@ -30,12 +31,13 @@ class CodeClient:
     def __init__(self):
         self._thread_pool = ThreadPoolExecutor(thread_name_prefix="evaluators_thread")
 
-    def _calculate_metric(self, evaluator, input_df, column_mapping, evaluator_name):
+    def _calculate_metric(self, evaluator, input_df, evaluator_name):
```

Comment:
Nice. Seems like the column mapping will happen ahead of time, and we are no longer relying on pf.run for column mapping. Is my understanding correct?

Reply:
Yes, essentially it is correct, however we are doing it only for thread-pool execution and for validation purposes. But yes, we essentially mimic this functionality in pandas way.